src/group_theory/group_action.lean

```diff
@@ -8,137 +8,158 @@ import data.set.finite group_theory.coset
 universes u v w
 variables {α : Type u} {β : Type v} {γ : Type w}
 
-class is_monoid_action [monoid α] (f : α → β → β) : Prop :=
-(one : ∀ a : β, f (1 : α) a = a)
-(mul : ∀ (x y : α) (a : β), f (x * y) a = f x (f y a))
+/-- Typeclass for types with a scalar multiplication operation, denoted `•` (`\bu`) -/
+class has_scalar (α : Type u) (γ : Type v) := (smul : α → γ → γ)
 
-namespace is_monoid_action
+infixr ` • `:73 := has_scalar.smul
 
-variables [monoid α] (f : α → β → β) [is_monoid_action f]
+class monoid_action (α : Type u) (β : Type v) [monoid α] extends has_scalar α β :=
+(one : ∀ b : β, (1 : α) • b = b)
+(mul : ∀ (x y : α) (b : β), (x * y) • b = x • y • b)
```

Comment:
How about renaming these to give them the same names as the version for modules, and generalizing all the module theorems that apply to monoid actions?